task/factory/default/src/main/java/io/knotx/fragments/task/factory/generic/node/action/ActionNodeFactory.java

```diff
@@ -59,69 +59,31 @@ public String getName() {
 
   @Override
   public ActionNodeFactory configure(JsonObject config, Vertx vertx) {
-    this.actionNameToOptions = new ActionNodeFactoryConfig(config).getActions();
-    Supplier<Iterator<ActionFactory>> actionFactoriesSupplier = () -> {
-      ServiceLoader<ActionFactory> factories = ServiceLoader.load(ActionFactory.class);
-      return factories.iterator();
-    };
-    actionProvider = new ActionProvider(actionFactoriesSupplier, actionNameToOptions, vertx);
+    Map<String, ActionFactoryOptions> actionNameToOptions = new ActionNodeFactoryConfig(config).getActions();
+    this.actionMetadataProvider = ActionMetadataProvider.create(NAME, actionNameToOptions);
+    this.actionProvider = new ActionProvider(SPI_ACTION_SUPPLIER, actionNameToOptions, vertx);
```

Comment:
`ActionProvider`'s constructor could actually take `Supplier<Iterable<ActionFactory>>` rather than `Supplier<Iterator<ActionFactory>>` - would simplify tests also